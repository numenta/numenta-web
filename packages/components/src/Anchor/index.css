/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.anchor {
  composes: clip from 'tachyons';  /* visibility: hidden; */
  composes: db from 'tachyons';
  composes: h3 from 'tachyons';
  margin-top: -4rem;  /* (tachyons h3) */
  z-index: -9999;
}
