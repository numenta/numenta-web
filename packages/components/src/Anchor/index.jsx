// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


const Anchor = ({name}) => (
  <span className={styles.anchor} id={name} />
)

Anchor.propTypes = {
  name: React.PropTypes.string.isRequired,
}

export default Anchor
