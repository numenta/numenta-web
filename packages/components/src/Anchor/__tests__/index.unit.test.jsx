import React from 'react'
import renderer from 'react-test-renderer'

import Anchor from '../../Anchor'


describe('Anchor React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Anchor name="technology" />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
