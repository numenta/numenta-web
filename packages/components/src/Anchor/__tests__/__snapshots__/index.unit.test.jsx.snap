exports[`Anchor React component Renders correctly 1`] = `
<span
  className="anchor"
  id="technology" />
`;
